server/src/main/java/org/apache/lucene/codecs/lucene90/Lucene90DocValuesConsumerWrapper.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.apache.lucene.codecs.lucene90;
+
+import org.apache.lucene.codecs.DocValuesConsumer;
+import org.apache.lucene.index.SegmentWriteState;
+
+import java.io.IOException;
+
+/**
+ * This class is an abstraction of the {@link DocValuesConsumer} for the Star Tree index structure.
+ * It is responsible to consume various types of document values (numeric, binary, sorted, sorted numeric,
+ * and sorted set) for fields in the Star Tree index.
+ *
+ * @opensearch.experimental
+ */
+public class Lucene90DocValuesConsumerWrapper {
```

Comment:
Let's please put these in a separate lib outside of server module.

Reply:
And also  `Lucene90DocValuesConsumer` is package protected , hence we need to place this class in `org.apache.lucene.codecs.lucene90` as well

Reply:
> And also `Lucene90DocValuesConsumer` is package protected , hence we need to place this class in `org.apache.lucene.codecs.lucene90` as well

Hmm we would need to pull the entire package out to /lib then which would be possible with 3.0?  Are there components we could pull?

>What advantage do you foresee with abstracing this out in a lib given we already have an org.apache.lucene java package within OpenSearch server module

My thinking is to start putting things in /lib that don't have anything to do with running a server.  Refactoring later on to pull things out is difficult and causes pain. I understand we would not want to make those breaking changes in 2.x line but we should be ok in 3.0?  You can have a pkg in /lib

Reply:
@mch2 should this be a blocker? We can create an issue for now, and maybe discuss the approach with the wider forum and take actions based on community feedback?

Reply:
@mch2 thanks for the suggestion, have created an [issue](https://github.com/opensearch-project/OpenSearch/issues/15317) to discuss this with a broader forum.